host/4/dotnet-build.yml

```diff
@@ -27,231 +27,122 @@ variables:
-                  -f host/4/bullseye/amd64/dotnet/dotnet8-isolated/dotnet-isolated-slim.Dockerfile \
-                  host/4/bullseye/amd64/dotnet/dotnet8-isolated/
-      npm run test $IMAGE_NAME --prefix  test/
-      docker push $IMAGE_NAME
-    displayName: dotnet-isolated-slim (.NET 8 Preview)
-    continueOnError: false
-
-  - bash: |
-      set -e
-      IMAGE_NAME=azurefunctions.azurecr.io/azure-functions/4/dotnet-isolated:$(Build.SourceBranchName)-dotnet-isolated8.0-appservice
-      IMAGE_ARRAY="$(image_list),$IMAGE_NAME"
-      echo "##vso[task.setvariable variable=image_list;]$IMAGE_ARRAY"
-
-      docker build -t $IMAGE_NAME \
-                  -f host/4/bullseye/amd64/out/dotnet/dotnet8-isolated-appservice.Dockerfile \
-                  host/4/bullseye/amd64/out/dotnet/
-      npm run test $IMAGE_NAME --prefix  test/
-      docker push $IMAGE_NAME
-
-      docker system df -v
-    displayName: dotnet-isolated-appservice (.NET 8 Preview)
-    continueOnError: false
-
-  - bash: |
-      docker system df -v
-      set -e
-      IMAGE_NAME=azurefunctions.azurecr.io/azure-functions/4/dotnet:$(Build.SourceBranchName)-dotnet6-bookworm
-      IMAGE_ARRAY="$(image_list),$IMAGE_NAME"
-      echo "##vso[task.setvariable variable=image_list;]$IMAGE_ARRAY"
-
-      docker build -t $IMAGE_NAME \
-                  -f host/4/bookworm/dotnet/dotnet6.Dockerfile \
-                  host/4/bookworm/dotnet
-      npm run test $IMAGE_NAME --prefix  test/
-      docker push $IMAGE_NAME
-    displayName: dotnet6-bookworm
-    continueOnError: false
-
-
-  - task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
-    displayName: 'Manifest Generator'
-    inputs:
-      BuildDropPath: host
-      Verbosity: Verbose
-      DockerImagesToScan: $(image_list)
-
-  - task: PublishPipelineArtifact@1
-    displayName: 'Publish SBOM Artifact'
-    inputs:
-      targetPath: host/_manifest
-      artifact: SBOM-Functionsv4-dotnet-$(Build.SourceBranchName)
\ No newline at end of file
+jobs:
+  - job: Dotnet
+    strategy:
+      matrix:
+        _inproc:
+          dotnetVersion: "dotnet"
+          dockerfilePath: "host/4/bullseye/amd64/dotnet/dotnet-inproc/"
+          dockerImagePath: "dotnet"
+          dockerImageName: ""
+        _inproc-slim:
+          dotnetVersion: "dotnet-slim"
+          dockerfilePath: "host/4/bullseye/amd64/dotnet/dotnet-inproc/"
+          dockerImagePath: "dotnet"
+          dockerImageName: "-slim"
+        _inproc-appservice:
+          dotnetVersion: "dotnet-inproc-appservice"
+          dockerfilePath: "host/4/bullseye/amd64/out/dotnet/"
+          dockerImagePath: "dotnet"
+          dockerImageName: "-appservice"
+        _isolated-net-6:
+          dotnetVersion: "dotnet-isolated"
+          dockerfilePath: "host/4/bullseye/amd64/dotnet/dotnet-isolated/"
+          dockerImagePath: "dotnet-isolated"
+          dockerImageName: "-dotnet-isolated-6.0"
```

Comment:
This name is different than the Dotnet Isolated Image name please ensure they remain the same

Reply:
I think if you fix this name then the tests should work. So the tests are kinda working as intended stopping us from releasing differently named images :D